client/src/main/java/io/pravega/client/SynchronizerClientFactory.java

```diff
@@ -36,9 +36,15 @@
      * @return Instance of ClientFactory implementation.
      */
     static SynchronizerClientFactory withScope(String scope, ClientConfig config) {
-        val connectionFactory = new SocketConnectionFactoryImpl(config);
-        return new ClientFactoryImpl(scope, new ControllerImpl(ControllerImplConfig.builder().clientConfig(config).build(),
-                connectionFactory.getInternalExecutor()), connectionFactory);
+        // Change the max number of number of allowed connections to the segment store to 1.
+        val updatedConfig = config.toBuilder()
```

Comment:
I don't think this change is appropriate. Here a user has a setting in the config that they can set, and we are just overriding it to be 1. Given that a factory can be used to create many objects there ought to be some way to configure the number of connections. This removes that.

I think a better solution would be to change the ClientConfig object to track if the value set is the default. Then if it has not been set then we can make the default different for different paths, such as by setting 1 here.

Reply:
Thanks, modified the code to ensure `ClientConfig `object is modified only if the default value is used.